src/osd/OSDMap.cc

```diff
@@ -4407,6 +4408,19 @@ void OSDMap::check_health(health_check_map_t *checks) const
 	d.detail.push_back(ss.str());
       }
     }
+
+    vector<int> destroyed;
+    get_destroyed_osds(&destroyed);
+    if (!destroyed.empty()) {
+      ostringstream ss;
+      ss << destroyed.size() << " osds destroyed";
+      auto& d = checks->add("OSD_DESTROYED", HEALTH_WARN, ss.str());
+      for (auto osd : destroyed) {
+        ostringstream ss;
+        ss << "osd." << osd << " is destroyed";
+        d.detail.push_back(ss.str());
+      }
+    }
```

Comment:
I'm not sure we need a warning here.  Just like a down + out osd is not an error (or warning) condition.  It's clutter in the map but harmless and not dangerous.